test-d/partial-on-undefined-deep.ts

```diff
@@ -1,4 +1,4 @@
-import {expectAssignable} from 'tsd';
+import {expectAssignable, expectType} from 'tsd';
 import type {PartialOnUndefinedDeep} from '../index';
 
 type TestingType = {
```

Comment:
because changed the key type of object in this PR, so it would be better to check the property keyword like `readonly`

```ts
readonly readonlyProperty: string | undefined;
```